### PR TITLE
Implement quick fix for eta going out of scope

### PR DESCRIPTION
**Pull request by ccahoughton**
#### PR description:

Added quick fix to eta bin indices going out of scope. [cms-sw/cmssw#43723](https://github.com/cms-sw/cmssw/issues/43723) shows that in CMSSW_14_0_0_pre2, a problem emerges when tracks with |eta| > 2.4 are put through the emulator, causing the resultant bin index to have a negative value. The quick fix is to simply discard tracks outside of the allowed eta region for the track jet emulator. There should be no change to the output jets. 

#### PR validation:

https://github.com/cms-sw/cmssw/issues/43723#issuecomment-1920655663 this comment shows a specific event that causes a segfault. With the implemented changes, there is no segfault at this event.

### CODE REVIEW

**Comment by epalencia** [comment]
Please test

**Comment by subirsarkar** [comment]
@ccahoughton Even if it is a quick fix, please use phiBins_ - 1 and etaBins_ - 1 in place of 23 and 26 respectively.

File: L1Trigger/L1TTrackMatch/plugins/L1TrackJetEmulatorProducer.cc [hunk 1]
```diff
@@ -297,6 +297,12 @@ void L1TrackJetEmulatorProducer::produce(Event &iEvent, const EventSetup &iSetup
       // Eta bin
       int j = eta_bin_firmwareStyle(L1TrkPtrs_[k]->getTanlWord());  //Function defined in L1TrackJetClustering.h
 
+      //This is a quick fix to eta going outside of scope - also including protection against phi going outside
+      //of scope as well. The eta index, j, cannot be less than zero or greater than 23 (the number of eta bins).
+      //The phi index, i, cannot be less than zero or greater than 26 (the number of phi bins).
+      if ((j < 0) || (j > 23) || (i < 0) || (i > 26))
```
Comment (srimanob) [inline L1Trigger/L1TTrackMatch/plugins/L1TrackJetEmulatorProducer.cc:303]: You allow bin=0, is it underflow? Or this is just your definition. I see you have the config as
```
etaBins=cms.int32(24),
phiBins=cms.int32(27),
```

**Comment by srimanob** [comment]
Testing in private production, issue is gone. I can completely produce sample with no crash (using 1000 events/lumi).

**Comment by epalencia** [comment]
@ccahoughton, could you please check the code-format error mentioned above?

**Comment by srimanob** [comment]
@cmsbuild please test

**Comment by epalencia** [comment]
+l1

**Comment by srimanob** [comment]
@ccahoughton 
As master is moved to 14_1, could you please make the backport of this PR to 14_0?

**Comment by srimanob** [comment]
Kindly ping @subirsarkar 
Thx.

**Comment by subirsarkar** [comment]
+Upgrade

**Comment by rappoccio** [comment]
+1

**Pull request merged**